openpype/pipeline/expected_files.py

```diff
@@ -0,0 +1,297 @@
+    families = set(
+        instance.data.get("families", []) + [instance.data["family"]]
+    )
+
+    tags = []
+    if reviewable:
+        tags.append("review")
+
+    # prepare base representation data
+    representation = {
+        "name": file_ext,
+        "ext": file_ext,
+        "stagingDir": output_dir,
+        # QUESTION: should we use persistent staging dir always?
+        "stagingDir_persistent": True,
+        "tags": tags,
+    }
+
+    # add frame range data
+    if frame_start is not None:
+        representation["frameStart"] = frame_start
+    if frame_end is not None:
+        representation["frameEnd"] = frame_end
+
+    # collect files from sequential path or single file in list
+    collected_file_frames = \
+        collect_files_from_sequential_path(
+            file_path, frame_start, frame_end, only_existing)
+
+    # set slate frame
+    if (
+        "slate" in families
+        # make sure frame range is set
+        and _add_slate_frame_to_collected_frames(
+            collected_file_frames, frame_start, frame_end)
+    ):
+        log.info("Slate frame added to collected frames.")
+        representation["frameStart"] = frame_start - 1
+
+    if len(collected_file_frames) == 1:
+        representation["files"] = collected_file_frames.pop()
+    else:
+        representation["files"] = collected_file_frames
+
+    return representation
+
+
+def _add_slate_frame_to_collected_frames(
+    collected_file_frames,
+    frame_start=None,
+    frame_end=None,
+):
+    """Add slate frame to collected frames.
+
+    Args:
+        collected_file_frames (list[str]): collected file frames
+        frame_start (Optional[int]): first frame
+        frame_end (Optional[int]): last frame
+
+    Returns:
+        Bool: True if slate frame was added
+    """
+    if (
+        frame_start is None
+        and frame_end is None
+    ):
+        return False
+
+    frame_start_str = frames.get_frame_start_str(
+        frame_start, frame_end)
+    frame_length = int(frame_end - frame_start + 1)
+
+    # add slate frame only if it is not already in collected frames
+    if frame_length == len(collected_file_frames):
+        frame_slate_str = frames.get_frame_start_str(
+            frame_start - 1,
+            frame_end
+        )
+
+        slate_frame = collected_file_frames[0].replace(
+            frame_start_str, frame_slate_str)
+        collected_file_frames.insert(0, slate_frame)
+
+        return True
+
+
+def get_single_filepath_from_list_of_files(collected_files):
+    """Get single filepath from list of files.
+
+    Args:
+        collected_files (list[str]): list of files
+
+    Returns:
+        Any[str, None]: single filepath or None if not possible
+    """
+    collections, reminders = clique.assemble(collected_files)
+    if collections:
+        return collections[0].format("{head}{padding}{tail}")
+    elif reminders:
+        return reminders[0]
+
```

Comment:
Note that there's nothing ensuring in this logic that there aren't more than one collection or more than one remainder - so it might be obfuscating/hiding potential bugs?